boards/st/stm32l562e_dk/stm32l562e_dk_common.dtsi

```diff
@@ -42,6 +44,81 @@
 	};
 };
 
+&fmc {
+	pinctrl-0 = <&fmc_nbl0_pe0 &fmc_nbl1_pe1
+		     &fmc_nce_pd7 &fmc_nwe_pd5 &fmc_noe_pd4
+		     &fmc_a0_pf0 &fmc_a1_pf1 &fmc_a2_pf2 &fmc_a3_pf3
+		     &fmc_a4_pf4 &fmc_a5_pf5 &fmc_a6_pf12 &fmc_a7_pf13
+		     &fmc_a8_pf14 &fmc_a9_pf15 &fmc_a10_pg0 &fmc_a11_pg1
+		     &fmc_a12_pg2 &fmc_a13_pg3 &fmc_a14_pg4 &fmc_a15_pg5
+		     &fmc_a16_pd11 &fmc_a17_pd12 &fmc_a18_pd13 &fmc_a19_pe3
+		     &fmc_a20_pe4 &fmc_a21_pe5
+		     &fmc_d0_pd14 &fmc_d1_pd15 &fmc_d2_pd0 &fmc_d3_pd1
+		     &fmc_d4_pe7 &fmc_d5_pe8 &fmc_d6_pe9 &fmc_d7_pe10
+		     &fmc_d8_pe11 &fmc_d9_pe12 &fmc_d10_pe13 &fmc_d11_pe14
+		     &fmc_d12_pe15 &fmc_d13_pd8 &fmc_d14_pd9 &fmc_d15_pd10>;
```

Comment:
Normally there is no need to list every fmc related pin, shouldn't it be?
```suggestion
pinctrl-0 = <&fmc_a0_pf0 &fmc_nce_pd7 &fmc_nwe_pd5 &fmc_noe_pd4
		     &fmc_d0_pd14 &fmc_d1_pd15 &fmc_d2_pd0 &fmc_d3_pd1
		     &fmc_d4_pe7 &fmc_d5_pe8 &fmc_d6_pe9 &fmc_d7_pe10
		     &fmc_d8_pe11 &fmc_d9_pe12 &fmc_d10_pe13 &fmc_d11_pe14
		     &fmc_d12_pe15 &fmc_d13_pd8 &fmc_d14_pd9 &fmc_d15_pd10>;
```

Shouldn't backlight `LCD_BL_CTRL.` (_PE1_) `fmc_nbl1_pe1` be controlled independently?

The same goes for `power-gpios`, that tends to be controlled using existing `regulator` driver.

Reply:
For the pin control, to be honest I used the one I saw on another .dtsi. Your solution seems to make more sense, so I fixed it.

For the backlight, I realized I never did anything with it. After some searching it seems like the backlight is always automatically on, even when not using the driver (even without this pull request). So I think I can just remove it from the pin-ctrl, and that's it ? Or should I turn it on more explicitly ?

For the `power-gpios`, I was looking at the regulator subsystem (I didn't know it existed) and found this in the doc :

> The simplest case where a regulator is needed is one where there is only one client. For those situations the cost of using the regulator device infrastructure is not justified, and *-gpios devicetree properties should be used. There is no device interface to these regulators as they are entirely controlled within the driver for the corresponding node, e.g. a sensor.

From my understanding this is more or less what I did ? If that isn't the case what exactly should I do ? I didn't find any trace of a regulator linked to the SoC/board, but maybe I didn't understood how it's supposed to work.

Reply:
> For the backlight, I realized I never did anything with it. After some searching it seems like the backlight is always automatically on, even when not using the driver (even without this pull request). So I think I can just remove it from the pin-ctrl, and that's it ? Or should I turn it on more explicitly ?

It can be left out if it is always on by default. There is no need to turn it on explicitly or add to fmc/display drivers.
A user can control it manually in an application if there is such a need.

> From my understanding this is more or less what I did ? If that isn't the case what exactly should I do ? I didn't find any trace of a regulator linked to the SoC/board, but maybe I didn't understood how it's supposed to work.

I'm not sure which way is the best or preferred.

Currently, it is possible to set a pin to the desired value using a `gpio-hog` or `fixed-regulator`. In that case, there is no need to add `power-gpios` to the driver's DT.

@erwango what is the preferred way?

Reply:
`regulator-fixed` is probably the way to go indeed as it won't require additional code.